Content.Server/Procedural/DungeonJob/DungeonJob.OreDunGen.cs

```diff
@@ -20,44 +21,55 @@ private async Task PostGen(
     {
         // Doesn't use dungeon data because layers and we don't need top-down support at the moment.
 
-        var emptyTiles = false;
         var replaceEntities = new Dictionary<Vector2i, EntityUid>();
         var availableTiles = new List<Vector2i>();
+        var tiles = _maps.GetAllTilesEnumerator(_gridUid, _grid);
 
-        foreach (var node in dungeon.AllTiles)
+        // WARNING:
+        // This DunGen handles not only the tiles of the passed dungeon, but ALL the tiles of the current grid
+        // So don't run it anywhere
+        while (tiles.MoveNext(out var tileRef))
         {
-            // Empty tile, skip if relevant.
-            if (!emptyTiles && (!_maps.TryGetTile(_grid, node, out var tile) || tile.IsEmpty))
-                continue;
-
-            // Check if it's a valid spawn, if so then use it.
-            var enumerator = _maps.GetAnchoredEntitiesEnumerator(_gridUid, _grid, node);
-            var found = false;
-
-            // We use existing entities as a mark to spawn in place
-            // OR
-            // We check for any existing entities to see if we can spawn there.
-            while (enumerator.MoveNext(out var uid))
-            {
-                // We can't replace so just stop here.
-                if (gen.Replacement == null)
-                    break;
+            var tile = tileRef.Value.GridIndices;
```

Comment:
My issue with changing these is that if someone wants infinite terrain then they just can't use this layer at all because it will bulldoze a lot of stuff. Dungeon layers really should only be affecting the dungeon tiles.

As is this layer would just break expeditions entirely because it would continuously re-generate the entire grid even if we only wanted to generate 1 tile.